crates/forge/src/scarb.rs

```diff
@@ -36,23 +41,70 @@ pub fn build_contracts_with_scarb(filter: PackagesFilter) -> Result<()> {
 
 pub fn build_test_artifacts_with_scarb(filter: PackagesFilter) -> Result<()> {
     ScarbCommand::new_with_stdio()
-        .arg("snforge-test-collector")
+        .arg("build")
+        .arg("--test")
         .packages_filter(filter)
         .run()
         .context("Failed to build test artifacts with Scarb")?;
     Ok(())
 }
 
+/// collecting by name allow us to dedup targets
+/// we do it because they use same sierra and we display them without distinction anyway
+fn test_targets_by_name(package: &PackageMetadata) -> HashMap<String, &TargetMetadata> {
+    fn test_target_name(target: &TargetMetadata) -> String {
+        // this is logic copied from scarb
+        target
+            .params
+            .get("group-id") // by unit tests grouping
+            .and_then(|v| v.as_str())
+            .map(ToString::to_string)
+            .unwrap_or(target.name.clone()) // else by integration test name
+    }
+
+    package
+        .targets
+        .iter()
+        .filter(|target| target.kind == "test")
+        .map(|target| (test_target_name(target), target))
+        .collect()
+}
+
 pub fn load_test_artifacts(
-    snforge_target_dir_path: &Utf8Path,
-    package_name: &str,
+    target_dir: &Utf8Path,
+    package: &PackageMetadata,
 ) -> Result<Vec<TestTargetRaw>> {
-    let test_artifacts_path =
-        snforge_target_dir_path.join(format!("{package_name}.snforge_sierra.json"));
+    let mut targets = vec![];
+
+    let dedup_targets = test_targets_by_name(package);
+
+    for (target_name, target) in dedup_targets {
+        let tests_location =
+            if target.params.get("test-type").and_then(|v| v.as_str()) == Some("unit") {
+                TestTargetLocation::Lib
+            } else {
+                TestTargetLocation::Tests
+            };
+
+        let target_file = format!("{target_name}.test.sierra.json");
+
+        match read_to_string(target_dir.join(target_file)) {
```

Comment:
Context: `cairo-profiler` expects paths to `VersionedProgram` (sierra format) in its input file. Right now `snforge` saves relevant sierra programs to separate files when using `--save-trace-data` or `--build-profile` and then puts paths to these files in the input file for `cairo-profiler`. This is a leftover from using `scarb-snforge-test-collector`
 
With this PR and Scarb 2.7.0 the sierra files with compiled tests are already present on the disk in the form `cairo-profiler` expects. Can we make `snforge` reuse the paths to these files when building input files for `cairo-profiler`? @ksew1 @cptartur 

Can be done in a separate PR

Relevant code:
- saving sierra program
https://github.com/foundry-rs/starknet-foundry/blob/4a1aea0f9f888c9b4774af897fc010d029d0180c/crates/forge/src/run_tests/test_target.rs#L42

- passing the path to the saved sierra program around:
https://github.com/foundry-rs/starknet-foundry/blob/4a1aea0f9f888c9b4774af897fc010d029d0180c/crates/forge-runner/src/running.rs#L85
to later use it here when generating input file for cairo-profiler
https://github.com/foundry-rs/starknet-foundry/blob/4a1aea0f9f888c9b4774af897fc010d029d0180c/crates/forge-runner/src/build_trace_data.rs#L41

Reply:
@ksew1 will you create an issue ?🙏

Reply:
On it

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/2355